catalog/core/catalog-core-impl/pubsub/src/main/java/ddf/catalog/pubsub/PublishedEventHandler.java

```diff
@@ -47,84 +55,99 @@ public PublishedEventHandler(Predicate finalPredicate, Subscription subscription
-
-        // if(this.catalog instanceof FanoutCatalogFramework)
-        // {
-        // // if the subscription is an enterprise subscription then evaluate all incoming events
-        // LOGGER.debug("Catalog framework is in fanout configuration - always evaluate events");
-        // evaluateEvent(event);
-        // }
-        // else if(this.subscription.isEnterprise())
-        if (this.subscription.isEnterprise()) {
-            // if the subscription is an enterprise subscription then evaluate all incoming events
-            LOGGER.debug("subscription is an enterprise subscription");
-            evaluateEvent(event);
-        } else if (sourceIds == null || sourceIds.isEmpty()) {
-            LOGGER.debug("subscription is a local subscription. Local Source Id: {}",
-                    catalog.getId());
-            if (catalog.getId() != null && catalog.getId()
-                    .equals(metacardSourceId)) {
-                LOGGER.debug("event received from local site");
-                evaluateEvent(event);
-            } else {
-                LOGGER.debug(
-                        "event is from remote site but subscription is local - not evaluating event against subscription filter");
+        @Override
+        public void run() {
+
+            String methodName = "handleEvent";
+            LOGGER.debug("ENTERING: {}", methodName);
+
+            if (LOGGER.isDebugEnabled()) {
+                LOGGER.debug("topic = {}", event.getTopic());
+                for (String propertyName : event.getPropertyNames()) {
+                    LOGGER.debug("name = {},   value = {}",
+                            propertyName,
+                            event.getProperty(propertyName));
+                }
+            }
+
+            if (subscription.getDeliveryMethod() instanceof Statusable
+                    && !((Statusable) subscription.getDeliveryMethod()).isAvailable()) {
+                LOGGER.debug("Subscription is not active ignoring event");
+                return;
             }
-        } else if (!sourceIds.isEmpty()) {
-            LOGGER.debug("subscription is a site-based subscription starting with site id {}",
-                    sourceIds.iterator()
-                            .next());
-            // perform site based filtering on subscription
-            if (sourceIds.contains(metacardSourceId)) {
-                LOGGER.debug("event received from subscribed site");
+
+
+            LOGGER.debug("subscription is enterprise? {}", subscription.isEnterprise());
+            Set<String> sourceIds = subscription.getSourceIds();
+            LOGGER.debug("subscription has source names: {}", sourceIds);
+
+            Metacard eventMetacard = (Metacard) event.getProperty(PubSubConstants.HEADER_ENTRY_KEY);
+            String metacardSourceId = eventMetacard.getSourceId();
+            LOGGER.debug("metacard source id: {}", metacardSourceId);
+
+
+            if (subscription.isEnterprise()) {
```

Comment:
Instead of having this cascade of `if/else if`, could we have a base `PublishedEventHandler`, three sub-classes with different `evaluateEvent()` implementations, and a factory class that creates the proper sub-class based on the conditions we have here?